lib/mayaUsd/ufe/UsdPointInstanceUndoableCommands.h

```diff
@@ -0,0 +1,216 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef MAYAUSD_UFE_USD_POINT_INSTANCE_UNDOABLE_COMMANDS_H
+#define MAYAUSD_UFE_USD_POINT_INSTANCE_UNDOABLE_COMMANDS_H
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/ufe/UsdPointInstanceOrientationModifier.h>
+#include <mayaUsd/ufe/UsdPointInstancePositionModifier.h>
+#include <mayaUsd/ufe/UsdPointInstanceScaleModifier.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+#include <mayaUsd/ufe/Utils.h>
+
+#include <pxr/base/gf/quath.h>
+#include <pxr/base/gf/vec3f.h>
+#include <pxr/usd/usd/timeCode.h>
+
+#include <ufe/path.h>
+#include <ufe/transform3dUndoableCommands.h>
+#include <ufe/types.h>
+
+#include <memory>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+/// Templated base class for undoable commands that manipulate USD point instances.
+///
+template <class BaseUfeUndoableCommand, class PointInstanceModifierType, class UsdValueType>
+class MAYAUSD_CORE_PUBLIC UsdPointInstanceUndoableCommandBase : public BaseUfeUndoableCommand
+{
+public:
+    using Ptr = std::shared_ptr<UsdPointInstanceUndoableCommandBase>;
+
+#ifdef UFE_V2_FEATURES_AVAILABLE
```

Comment:
If UsdTransform3dPointInstance depends on UFE_V2, does it make sense to conditionally compile in UsdPointInstanceUndoableCommands for UFE_V1?

Reply:
Good catch. I admittedly have only been working with this code against PR 122, so there's an excellent chance this doesn't even build successfully against Maya 2020 as it is right now.

I think my intent was to gracefully ignore attempts to manipulate point instances for UFE v1. The way things are currently setup, it should be possible in UFE v1 to have UFE selections that include scene items representing point instances, even though point instance selection in the viewport is UFE v2 only. In that case, I was aiming for requests to manipulate point instance items in UFE v1 basically becoming no-ops. That was why I had the modifiers and the commands included in the CMakeLists.txt for *both* UFE v1 and UFE v2.

So I think what I actually have to do is ensure that all of the new files here are compiled for both UFE versions. Does all of this sound right/reasonable to you?

Reply:
If UFE v1 is intended to be a no-op, then it seems to me that you could just return null pointers when you're asked for commands.  That way, you wouldn't need to compile commands against UFE v1 --- and if the commands are the only clients of the modifiers, then the modifiers wouldn't need to be compiled against UFE v1 either.  Does this align with your thinking?

Reply:
It does, thanks! In the most recent changes, `UsdPointInstanceUndoableCommands.h` is only included for UFE v2.

I ended up leaving the modifiers in for both versions of UFE, since strictly speaking there's nothing in them that requires v2, so they could theoretically get used for non-Transform3d authoring operations.